### PR TITLE
Add authorization requests to unit class

### PR DESCRIPTION
**Pull request by obando777**
## Add authorization requests to unit class

allow client to fetch authorization requests, approve and decline, as it is described here https://docs.unit.co/#authorization-requests

TODO:
- address any comment over this PR

### CODE REVIEW

File: types/authorizationRequest.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+         * 	        DoNotHonor,
+         * 	        InsufficientFunds,
+         * 	        InvalidMerchant,
+         * 	        ReferToCardIssuer,
+         * 	        RestrictedCard,
+         * 	        Timeout,
+         * 	        TransactionNotPermittedToCardholder.
+         * Available only when status is Declined
+         */
+        declineReason?: string
```
Comment (axshani) [inline types/authorizationRequest.ts:56]: we are using this convention

```suggestion
        declineReason?:  "AccountClosed" | "CardExceedsAmountLimit" | ....
```

File: types/authorizationRequest.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+     * One of AccountClosed,
+     *        CardExceedsAmountLimit,
+     *        DoNotHonor,
+     *        InsufficientFunds,
+     *        InvalidMerchant,
+     *        ReferToCardIssuer,
+     *        RestrictedCard,
+     *        TransactionNotPermittedToCardholder.
+     */
+    reason: string
```
Comment (axshani) [inline types/authorizationRequest.ts:127]: see my previous suggestion

File: resources/authorizationRequest.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                reason: request.reason
+            }
+        }
+        return await this.httpPost<UnitResponse<PurchaseAuthorizationRequest>>(path, { data })
+    }
+}
+
+interface AuthorizationRequestQueryParams {
+    /**
+         * Maximum number of resources that will be returned. Maximum is 1000 resources. [See Pagination](https://developers.unit.co/#intro-pagination).
```
Comment (axshani) [inline resources/authorizationRequest.ts:55]: indentions
we recommend using ESLint

**Review by axshani — CHANGES_REQUESTED** [review]
Nice one!

File: types/authorizationRequest.ts (hunk 1 above)
Comment (obando777) [reply types/authorizationRequest.ts:56]: thank you on your quick response, i will do it

**Comment by obando777** [comment]
> Nice one!

thank you for your amazing work guys.  let me know about any other comment

File: resources/authorizationRequest.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        const parameters = {
+            "page[limit]": (params?.limit ? params?.limit : 100),
+            "page[offset]": (params?.offset ? params?.offset : 0),
+            ...(params?.accountId && { "filter[accountId]": params?.accountId }),
+            ...(params?.customerId && { "filter[customerId]": params?.customerId })
+        }
+
+        return this.httpGet<UnitResponse<PurchaseAuthorizationRequest[]>>("", { params: parameters })
+    }
+    public async approve(request: ApproveAuthorizationRequest): Promise<UnitResponse<PurchaseAuthorizationRequest> | UnitError> {
```
Comment (Elroie) [inline resources/authorizationRequest.ts:30]:
```suggestion

    public async approve(request: ApproveAuthorizationRequest): Promise<UnitResponse<PurchaseAuthorizationRequest> | UnitError> {
```

**Review by Elroie — APPROVED** [review]

**Review by Elroie — APPROVED** [review]

**Review by axshani — APPROVED** [review]

**Pull request merged**